crates/templates/src/writer.rs

```diff
@@ -47,7 +48,110 @@ impl TemplateOutput {
                     .await
                     .with_context(|| format!("Can't save changes to {}", path.display()))?;
             }
+            TemplateOutput::MergeToml(path, target, text) => {
+                let existing_toml = tokio::fs::read_to_string(path)
+                    .await
+                    .with_context(|| format!("Can't open {} to append", path.display()))?;
+                let new_toml = merge_toml(&existing_toml, target, text)?;
+                tokio::fs::write(path, new_toml)
+                    .await
+                    .with_context(|| format!("Can't save changes to {}", path.display()))?;
+            }
         }
         Ok(())
     }
 }
+
+fn merge_toml(existing: &str, target: &str, text: &str) -> anyhow::Result<String> {
+    use toml_edit::{Document, Entry, Item};
+
+    let mut doc: Document = existing
+        .parse()
+        .context("Can't merge into the existing manifest - it's not valid TOML")?;
+    let merging: Document = text
+        .parse()
+        .context("Can't merge snippet - it's not valid TOML")?;
+    let merging = merging.as_table();
+    match doc.get_mut(target) {
+        Some(item) => {
+            let Some(table) = item.as_table_mut() else {
+                anyhow::bail!("Cannot merge template data into {target} as it is not a table");
+            };
+            for (key, value) in merging {
+                match table.entry(key) {
+                    Entry::Occupied(mut e) => {
+                        let existing_val = e.get_mut();
```

Comment:
What is the thought process on the added component's variables trumping existing component vars? I'd think the other way around would be preferred. Better to break new functionality than existing functionality IMO

Reply:
The thought process was roughly "if there is a clash, it's likely because I'm re-running the same template, in which case I probably want the new definitions rather than the old ones."  And if I do want to keep the old definitions then I can still refer back to them through source control.  That's not to say the thought process was right of course!